public_html/wp-content/themes/pattern-directory/single-wporg-pattern.php

```diff
@@ -9,7 +9,25 @@
 
 namespace WordPressdotorg\Theme;
 
+use const WordPressdotorg\Pattern_Directory\Pattern_Flag_Post_Type\POST_TYPE;
+
 get_header();
+
+$userHasReported = false;
+
+if( is_user_logged_in() ) {
+	$args = array(
+		'author' => get_current_user_id(),
+		'post_parent' => get_the_ID(),
+		'post_type' => POST_TYPE,
+		'post_status' => 'any'
+	);
+	
+	$items = new \WP_Query( $args );
+	$userHasReported = $items->have_posts();
+}
```

Comment:
I'm not sure where to place this code? @coreymckrill @ryelle @iandunn Should this be moved to a class in the plugin as opposed to having it in the theme? 

I think the Plugin Directory has a default class and is called everywhere for this like this.

Reply:
I can't think of a use case for this on the plugin side. But this could be moved to a function and put into the theme's **functions.php** file.

Reply:
A potentially naive part of me thinks that reducing the number of places that plugin-specific logic is used amounts to a better strategy over time but I'll move this logic to functions.php for now.

Reply:
I also noticed that the theme files (like this one) use the namespace `namespace WordPressdotorg\Theme` while the functions.php uses `namespace WordPressdotorg\Pattern_Directory\Theme`. Do you know if that's intentional?

Reply:
No idea about the namespace thing...

Reply:
Ahhh, I copied these from the wporg parent theme and didn't notice 😬  Fixed that in d8a68b6471d30b7eb34a1e1452602f659efde6b3